### PR TITLE
Hw1 task05

### PR DESCRIPTION
**Pull request by mirmozavr**

### CODE REVIEW

**Comment by Cjkjvfnby** [comment]
It works, now you can see the check UI.  

See my comment about complexity in the teams chat.

image

File: hw1/task05/task05.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    nums = [1, 3, -1, -3, 5, 3, 6, 7], k = 3
+    result = 16
+"""
+
+
+def find_maximal_subarray_sum(nums: list, k: int) -> int:
+    """Max sum of continuous sub-array."""
+    ans = -float("INF")
+    index = 0
+    n = len(nums)
```
Comment (Cjkjvfnby) [inline hw1/task05/task05.py]: Let's give it a meaningful name.

File: hw1/task05/task05.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    # fixing situation where k > N
+    k = min(n, k)
+    # find max subarray sum and its start index
+    for i in range(n - k + 1):
+        if sum(nums[i : i + k]) > ans:
+            ans = sum(nums[i : i + k])
+            index = i
+
+    # exclude negative numbers
+    for number in nums[index : index + k]:
```
Comment (Cjkjvfnby) [inline hw1/task05/task05.py]: `pytest.mark.parametrize` is a good thing to remove boilerplate, but it should be used with caution. 

You have grouped tests that cover different cases.   I'd recommend writing a separate for that particular test case.   

I have some doubts about this code.  How can we be sure that this is still max value, after reducing it?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: hw1/task05/task05.py (hunk 1 above)
Comment (mirmozavr) [reply hw1/task05/task05.py]: Agreed

File: hw1/task05/task05.py (hunk 2 above)
Comment (mirmozavr) [reply hw1/task05/task05.py]: We are checking if a number is negative. So double negation do increase max sum value.

Comment (mirmozavr) [reply hw1/task05/task05.py]: About tests, what do you have in mind?

File: hw1/task05/task05.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    """Max sum of continuous sub-array."""
+    yet_max_sum = 0
+    size = len(nums)
+    # fixing situation where k > N
+    k = min(size, k)
+
+    for i in range(size):
+        for j in range(1, k + 1):
+            yet_max_sum = max(sum(nums[i : i + j]), yet_max_sum)
+            if i + j >= size:
```
Comment (Cjkjvfnby) [inline hw1/task05/task05.py:22]: Why not embed this check to the range?

**Review by Cjkjvfnby — APPROVED** [review]

File: hw1/task05/task05.py (hunk 3 above)
Comment (mirmozavr) [reply hw1/task05/task05.py:22]: I didn't see the way to do it. Is there any?

Comment (Cjkjvfnby) [reply hw1/task05/task05.py:22]: You slice nums by specific indexes.  You can move calculation from slice to range.

Comment (mirmozavr) [reply hw1/task05/task05.py:22]: Solution found, branch merged

**Pull request merged**